generate_directory_indexes.py

```diff
@@ -115,47 +156,131 @@ def format_directory_listing(directory_listing):
     })
     return out
 
-def walk(path, base_path):
-    contents = os.listdir(path)
-    directory_listing = []
+def file_information(full_path, file, last_modified = None, size = None):
+    if last_modified == None:
+        last_modified = os.path.getmtime(full_path)
+    if size == None:
+        size =  os.path.getsize(full_path)
+    icon = 'folder.gif' if os.path.isdir(full_path) else 'unknown.gif'
+
+    return {
+        'name': file,
+        'lastModified': last_modified,
+        'size': size,
+        'icon': icon
+    }
+
+def string_to_epoch_seconds(string, format):
+    utc_time = datetime.strptime(string, format)
+    epoch_time = (utc_time - datetime(1970, 1, 1)).total_seconds()
+
+    return epoch_time
+
+
+def is_an_index_file(file_name):
+    # A list of file names to exclude from the index files.
+    index_file_names = [
+        "index.html",
+        "index_by_lastModified.html",
+        "index_by_lastModified_reverse.html",
+        "index_by_name.html",
+        "index_by_name_reverse.html",
+        "index_by_size.html",
+        "index_by_size_reverse.html"
+    ]
+
+    if file_name in index_file_names:
+        return True
+
+    return False
```

Comment:
couldn't you just `return file_name in index_file_names`?

Reply:
Wrong direction. The intent here is to provide a condition for not including certain files in the index.html files, namely the index.html files themselves. Previously there was a regex for that that I replaced with this whitelist. 

IOW, does current file match the special file? If Yes, then skip it.

A refactoring in the future will combine the logic of generating the index file name with creating the skippable-list. However, after some thought, that felt outside the scope of this already change-heavy PR.

Reply:
I understand how you're using this functions, but I'm pretty sure, if you zoom out, this just says
```
if (true)
   return True
return False
```
unless I'm just wrong about what python's `<blah> in <blah_blah>` does..

Reply:
file_name will rarely be in index_file_names. This function will return False most of the time.

Reply:
yeah, but my point is that this could just be simplified.
so sometimes, `return (true)`, but usually `return (false)`.
Am I making sense?

Reply:
I could use:

```python
return True if file_name in index_file_names else False
```
but I don't find that terribly readable.

Reply:
This will work:

```python
return file_name in index_file_names
```
That's likely what you meant. It is more succinct but isn't as obvious IMO.

Reply:
yeah, that's what I meant. There's definitely a balance between readability and succinctness, so I'll let you make the call.

Reply:
After thinking about it overnight your version, I think, is correct. There's another tweak that I want to try as well.